base/doc/clsguide.tex

```diff
@@ -977,6 +977,30 @@ \subsection{Passing options around}
 handler, but this handler is not used for |landscape| because that option is
 explicitly declared.
 
+\subsection{Testing dates of other code}
+
+\begin{decl}
+  |\IfPackageAtLeast(TF)| \arg{package} \arg{date} \arg{true code} \arg{false code}\\
+  |\IfClassAtLeast(TF)| \arg{class} \arg{date} \arg{true code} \arg{false code}\\
+  |\IfFileAtLeast(TF)| \arg{file} \arg{date} \arg{true code} \arg{false code}
+\end{decl}
+There are occasions when you need to test if a package is
+at least a particular date (assuming it has been loaded). This can be done using the
+commands |\IfPackageAtLeastTF|, etc. Separate commands are available to test
+packages, classes and general files: in all cases, they rely on the existence
+of a |\Provides...| line in the file being queried. Three versions
+of each command are available, ending |T|, |F| and |TF|, which indicate the
+outcome branches required. Thus for example |\IfPakcageAtLeastT| does not need
+a \meta{false code} argument. The \meta{date} argument can be given either in
+ISO (|YYYY-MM-DD|) or traditional \LaTeX{} (|YYYY/MM/DD|) formats.
+
+\begin{decl}
+  |\IfFormatAtLeast(TF)| \arg{date} \arg{true code} \arg{false code}
```

Comment:
```suggestion
  |\IfFormatAtLeastTF| \arg{date} \arg{true code} \arg{false code}
```

Also `\IfFormatAtLeastTF` doesn't match the title of current subsection
```tex
\subsection{Testing dates of other code}
```

Perhaps
```tex
\subsection{Testing dates of the \LaTeX{} format and \pkg{expl3}}
\IfFormatAtLeastTF
\IfExplAtLeastTF

\subsection{Testing dates of other code}
\IfPackageAtLeastTF
\IfClassAtLeastTF
\IfFileAtLeastTF
```

Reply:
> Well the kernel is 'other code' :)
> <sub>_Originally posted by @josephwright in https://github.com/latex3/latex2e/pull/1452#discussion_r1738885024_</sub>

Reply:
Not a bad idea!